### PR TITLE
feat: best-of-n jailbreak

### PR DESCRIPTION
**Pull request by typpo**

### CODE REVIEW

File: src/redteam/strategies/index.ts [hunk 1]
```diff
@@ -63,6 +64,15 @@ export const Strategies: Strategy[] = [
       return newTestCases;
     },
   },
+  {
+    id: 'best-of-n',
+    action: async (testCases, injectVar, config) => {
+      logger.debug('Adding Best-of-N to all test cases');
```
Comment (mldangelo) [inline src/redteam/strategies/index.ts]: nit for later - I know we support the concept of filtering strategies to specific plugins. I am not sure where that happens and if this message makes sense all the time.

File: examples/redteam-bestOfN/promptfooconfig.yaml
Comment (mldangelo) [inline examples/redteam-bestOfN/promptfooconfig.yaml:1]: consider adding the word strategy to this file if you want to keep this.

File: site/docs/red-team/strategies/best-of-n.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+---
+sidebar_label: Best-of-N
+---
+
+# Best-of-N (BoN) Jailbreaking Strategy
```
Comment (mldangelo) [inline site/docs/red-team/strategies/best-of-n.md:5]: excellent job here

File: src/redteam/strategies/index.ts (hunk 1 above)
Comment (typpo) [reply src/redteam/strategies/index.ts]: good point

**Pull request merged**